pkg/debug/checkups/quarantine.go

```diff
@@ -0,0 +1,228 @@
+package checkups
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+// quarantine:
+// Recursively scans common installation directories to to a given depth.
+// Reports and directories that have the word "quarantine" in their path and the number of files and their names they contain.
+// Fails if any files are found in the above directories.
+// Reports possible "meddlesome" processes for information purposes (does not fail due to proccesses running)
+
+// It's difficult to keep track of every possible Anti-Virus or EDRs quarantine directory, but they all seem
+// to have "quarantine" in their name. So we just look for that some where in the dir path. The suspicion
+// is that some programs will quarantine osquery. Unfortunalty, we typically can't see the names of the files
+// that were quarantined. So if we do find quarantined files, we'll fail and would ask the user to check and
+// see if osquery was quarantined.
+
+type quarantine struct {
+	status                     Status
+	summary                    string
+	quarantineDirPathFilenames map[string][]string
+	dirsChecked                int
+}
+
+func (q *quarantine) Name() string {
+	return "Quarantine"
+}
+
+func (q *quarantine) searchPathDepths() map[string]int {
+	switch runtime.GOOS {
+	case "windows":
+		return map[string]int{
+			// Crowdstrike: C:\Windows\System32\Drivers\CrowdStrike\Quarantine
+			`C:\Windows\System32\Drivers`: 3,
+			// Malwarebytes: C:\ProgramData\Malwarebytes\MBAMService\Quarantine
+			// Windows Defender: C:\ProgramData\Microsoft\Windows Defender\Quarantine
+			`C:\ProgramData`: 3,
+		}
+	case "darwin":
+		return map[string]int{
+			// Crowdstrike: /Library/Application Support/CrowdStrike/Falcon/Quarantine
+			`/Library/Application Support`: 4,
+		}
+	case "linux":
+		return map[string]int{
+			// Malwarebytes: /var/lib/mblinux/quarantine
+			`/var/lib`: 3,
+		}
+	default:
+		return make(map[string]int)
+	}
+}
+
+func (q *quarantine) Run(ctx context.Context, extraFh io.Writer) error {
+	q.quarantineDirPathFilenames = make(map[string][]string)
+
+	var (
+		meddlesomeProcessPatterns = []string{
+			`crowdstrike`,
+			`opswat`,
+			`defend`,
+			`defense`,
+			`threat`,
+			`virus`,
+			`quarantine`,
+			`snitch`,
+		}
```

Comment:
Do we know what carbonblack's process name is?

Reply:
googling revealed many possibilities, I added all the ones that looked promising